src/app/metadata/services/metadataSearch.service.ts

```diff
@@ -27,21 +28,23 @@ export class MetadataSearchService {
   #className = signal<string | undefined>(undefined);
   #limit = signal<number | undefined>(undefined);
   #skip = signal<number | undefined>(undefined);
+  #query = signal<string | undefined>(undefined);
+  #facets = signal<FacetFilterSetting>({});
+
+  facets = computed(() => this.#facets());
+  query = computed(() => {
+    if (!this.#query()) return '';
+    return this.#query();
+  });
 
   #searchResults = resource({
     request: computed(() => ({
-      className: this.#className(),
-      limit: this.#limit(),
-      skip: this.#skip(),
+      url: this.#massQueryUrl(),
     })),
     loader: (param) => {
```

Comment:
```suggestion
    loader: ({request}) => {
```
I find this a bit nicer than the obscure `param` attribute. And then you can just access request instead of param.request.

Reply:
fixed.